### ISSUE TITLE
"Removed unreachable peer" stalled network comms

### ISSUE THREAD

**Issue by richstokes** [weaveworks/weave#3651]
On `weave  2.5.0` we were seeing intermittent network/dns communication issues. Kubernetes `v1.11.9`

In the weave logs we saw lots of messages relating to "Removed unreachable peer".

(FWIW this is our test cluster, and we delete/add nodes on a regular basis. Some of the peers listed in the logs had been deleted previously.)

I found this issue relating to a prior version of weave: https://github.com/weaveworks/weave/issues/3384

Looks like the solution documented here is still applicable. Reporting as it still seems to be an issue with weave 2.5.0 (cluster installed via kops).

To confirm, the fix was:

ssh into the EC2 instances (masters and workers) one by one and delete the file /var/lib/weave/weave-netdata.db:

`sudo rm /var/lib/weave/weave-netdata.db`

restart all the weave pods by deleting them: 
```
for i in $(kubectl get pods -n kube-system | awk '{print $1}' | grep weave); do kubectl delete pod -n kube-system $i; done
```

**Comment by murali-reddy** [comment]
Message `"Removed unreachable peer"` is expected message when a node is deleted. Could you please elaborate the symptoms when you say "seeing intermittent network/dns communication issues"?

It would be helpful to share `weave status connections` and `weave status peers` also check logs for any errors.

**Comment by richstokes** [comment]
I'd have pods that couldn't talk to each other via their overlay/service IP's - I'd get errors like "No route to host".

I'll get that extra info if the problem returns, taking the above steps has fixed it. There were not any other obvious errors in the logs, just the "Removed unreachable peer" seemed to be repeating itself.

**Comment by bboreham** [comment]
@richstokes we can often spot what is wrong when you supply the logs and other information requested.

Hinting at what they said is of negative value.

**Comment by mariusv** [comment]
I'm seeing the same issue. As in the node to which the connection is deleted is still up and running and healthy but on all the wavenet pods logs I can see this:

```
kube-system/weave-net-ctjd4[weave]: INFO: 2019/11/29 17:10:15.362793 Removed unreachable peer 11x:xx:xx:xx:ab:9e(p-10-xxx-x-x.xxx-xxxxx-1.compute.internal)
kube-system/weave-net-rsxp8[weave]: INFO: 2019/11/29 17:10:15.361507 Removed unreachable peer 11x:xx:xx:xx:ab:9e(p-10-xxx-x-x.xxx-xxxxx-1.compute.internal)
kube-system/weave-net-7qzq9[weave]: INFO: 2019/11/29 17:10:15.362986 Removed unreachable peer 1x:xx:xx:xx:ab:9e(ip-10-xxx-x-x.xxx-xxxxx-1.compute.internal)
kube-system/weave-net-hsqpx[weave]: INFO: 2019/11/29 17:10:15.362441 Removed unreachable peer 11x:xx:xx:xx:ab:9e(p-10-xxx-x-x.xxx-xxxxx-1.compute.internal)
```

**Comment by bboreham** [comment]
(a) we never got enough information to debug
(b) some issues in this area were fixed in 2.6.

If you are still having problems please click 'new issue' and supply the requested information.

**Comment by TechnicalVegeta** [comment]
If you see the message: **IP allocation was seeded by different peers**, this means that some Weave Net peers were initialized into one cluster and some into another cluster; Weave Net cannot operate in this state.
To recover, you need to eliminate the IPAM data from the affected nodes and restart. If you installed via the Kubernetes Addon, this data will be in a file under /var/lib/weave on the node
